### ISSUE TITLE
severity literals

### ISSUE THREAD

**Issue by FND** [eslint/eslint#4046]
As I understand it, a rule's severity is currently expressed with a number (0/1/2) as the first item in the list of rule configuration options:

```yaml
rules:
    indent:
    - 2
    - 4
```

This can make `.eslintrc` somewhat cryptic (even misleading) for the uninitiated, so it would be nice to be able to use literals instead:

```yaml
rules:
    indent:
    - warning
    - 4
```

(I've tried extending [`getRuleSeverity`](https://github.com/eslint/eslint/blob/adf5124de692bf7f9915b82c7351c69b61f94741/lib/eslint.js#L573) with a simple mapping to convert literals to the corresponding number, but it seems I'd also need to adjust `config-validator`?)

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
See discussion on https://github.com/eslint/eslint/issues/3626.
